anti_patterns/deny-warnings.md

```diff
@@ -0,0 +1,35 @@
+# `#![deny(warnings)]`
+
+## Description
+
+A well-intentioned crate author wants to ensure their code builds without
+warnings. So they annotate their crate root with
+
+## Example
+
+```rust
+#![deny(warnings)]
+
+// All is well.
+```
+
+## Drawbacks
+
+Until, say, the language changes, and as part of the transition, a lint is
+introduced that `warn`s, and Boom! The code no longer builds. Or someone tries
+to compile the code with additional lints (e.g. [clippy]), only to find out
+that they have to remove the annotation first.
+
+## Alternatives
+
+Our plucky developer hero should have used
```

Comment:
You're right. I was a bit aggressive because I've had to deal with so many crates doing this, just creating an obstacle for me to test clippy on their code...it's all good now :smile: